src/server/routers/home-block.router.ts

```diff
@@ -20,4 +24,12 @@ export const homeBlockRouter = router({
     .use(isFlagProtected('alternateHome'))
     .use(applyUserPreferences())
     .query(getHomeBlocksByIdHandler),
+  createCollectionHomeBlock: protectedProcedure
+    .input(createCollectionHomeBlockInputSchema)
+    .use(isFlagProtected('alternateHome'))
+    .mutation(createCollectionHomeBlockHandler),
+  delete: protectedProcedure
+    .input(getByIdSchema)
+    .use(isFlagProtected('alternateHome'))
```

Comment:
We can probably add a middleware, right after this line, to check whether a user is able to perform this action by checking if they're the owner of the homeblock or a moderator. This way we prevent anyone from deleting entries from other people. WE have done it in other places, you can check for an example here https://github.com/civitai/civitai/blob/7ef7ced1279375888e2058b126e8d7ddd46c0ece/src/server/routers/image.router.ts#L34

Reply:
It's on the service level atm - but good thinking 🤔  Might also protect it here